### ISSUE TITLE
Data Views: Pages: Improve clarity of the Trash and Permanently delete flows

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60383]
### Description

See previous similar issue https://github.com/WordPress/gutenberg/issues/52639

From the Site editor > pages, a page can be 'moved to the trash'. Screenshot:

![01](https://github.com/WordPress/gutenberg/assets/1682452/ad80dc30-81d1-4a3a-b909-a017ffbebab7)

However, the confirm dialog that opens when trashing a page, asks users whether they want to _delete_ the page. Screenshot:

![02](https://github.com/WordPress/gutenberg/assets/1682452/9ac27e7e-efa2-4ccf-bdaa-6c0778b84267)

That's less than ideal. It's confusing for users, as they will not trust the editor and potentially fear the page is going to be _permantently deleted_.

After confirming the action, a snackbar notice will confirm the page has been 'moved to the trash'. Screenshot:

![03](https://github.com/WordPress/gutenberg/assets/1682452/3c3477fe-b5e6-42c9-9f80-fe479dc4ff11)

Once the page is in the trash, useres can permanently delete it. One of the available actions is 'Permanently delete'. Screenshot:

![04](https://github.com/WordPress/gutenberg/assets/1682452/5a7c53e3-6779-4778-b760-6b08fc4e305b)

After clicking Permanently delete' there is no confirmation step. The page is _permanently deleted_ right away. A snackbar notice will confirm the permanent deletion:

![05](https://github.com/WordPress/gutenberg/assets/1682452/2ca6cea0-678b-4250-b6da-620056c0354b)

Overall, I think this flow is unclear and confusing for users.
- I'm not sure whether the 'move to trash' action should go through a confirm dialog in the first place. That's not a destructive action. Anyways, the confirm dialog copy and the button text are wrong because they refer to 'delete'.
- The permanent deletion action doesn't go through a confirm step. This is a destructive action. I'd consider to add a confirm dialog to avoid data loss by accident.

## Some history
Traditionally, the classic admin never implemented confirmation dialogs for the 'trash page' and 'delete permanently page' actions. The reasoning was that the first action isn't a destructive one. The second action is desctructive but users are in the context of hte 'trash' and it's clear enough that deelting from the trash _deletes permanently_.

Regardless od the historical approach illustrated above, it is worth considering whether it would be beneficial to think at a better flow. My 2 cents:
- The trash action shouldn't have a confirm dialog.
- The permantently delete action should have it.

### Step-by-step reproduction instructions

- Go to the site editor > Pages > switch to the List view to make things more clear.
- Trash a page.
- Observe the confirmation dialog says 'delete'.
- Once the page is trashed, go to the Pages Trash section
- In the Actione menu for a page, click 'Delete permanently'.
- Observe there is no confirmation step.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
Just noting this as an overlapping issue with https://github.com/WordPress/gutenberg/issues/46859 and https://github.com/WordPress/gutenberg/issues/60072 

cc @jameskoster as we seem to keep running into each other in issues around this general topic :D

**Comment by ammar-mohamed-massoud** [comment]
I would love to work on that as this was confusing for me to with all my 5 years developing WordPress sites
We need to give it a new feel like the mentioned one 
So is this is a thing I can work on :)

**Comment by jameskoster** [comment]
Yes we went back and forth on page actions a bit, and it seems a regression snuck in somewhere. 

I agree this seems backwards – moving to trash should not invoke a confirmation dialog, permanently deleting should.

**Comment by joedolson** [comment]
I'm reopening this issue because the resolving PR wasn't complete; it added the confirmation before deletion, but didn't remove the confirmation when moving to trash. It's mostly complete, but not totally.

**Comment by shail-mehta** [comment]
Created a PR to remove the Trash Confirmation Dialog: https://github.com/WordPress/gutenberg/pull/82297

**Comment by oandregal** [comment]
We made a few changes since this issue was reported:

- no delete icon
- text in the action no longer refers to "delete"
- add ellipsis to communicate actions have 

This is the current status:

https://github.com/user-attachments/assets/a0ab10e9-9291-4d23-bbe9-1aa7dabf0729

> Created a PR to remove the Trash Confirmation Dialog: https://github.com/WordPress/gutenberg/pull/82297

By doing this, any _published_ post can be unpublished and sent to the trash, even if accidentally clicked. 

An improvement we could make is to:

- unpublished posts (scheduled, draft): send them to draft immediately
- published posts: keep the confirmation

We actually tried this at https://github.com/WordPress/gutenberg/pull/81994#issuecomment-5513617336 (cc @jasmussen @ciampo) but bulk actions become confusing (see details there). Perhaps the way forward here is to actually have two separate actions:

- unpublish: 
  - only available for published posts
  - executed with confirmation (modal)
- trash:
  - only available for unpublished posts (draft, scheduled)
  - executed immediately (no modal)

Thoughts?

**Comment by jasmussen** [comment]
For me the big discrepancy was the addition of the ellipsis to items that open modals, and no ellipsis on items that don't. This change on its own is perfectly valid, one I support, but when it becomes visible in the quick actions, it means as you mouse over pages some of which are drafts and others which are publishe, things jump around. If we're fine with that, cool.

**Comment by afercia** [comment]
> Thoughts?

Whatever the decision is, I'd strongly encourage to take ownership of the consistency issue. Changes to the trash flow should be ported to Core as well, to provide users with a consistent experience.

Right now on core:
- Published posts that get trashed don't use any confirmation step.
- Restoring from the Trash a post that was originally Published, moves it to Drafts first. It's not directly restored as 'Published'. This was changed a while ago for some reasons I don't fully remember now, but you get the point. 

Making sure the user experience is equivalent with Classic list tables and dataviews is key.

**Comment by oandregal** [comment]
> Restoring from the Trash a post that was originally Published, moves it to Drafts first. It's not directly restored as 'Published'. This was changed a while ago for some reasons I don't fully remember now, but you get the point.

This is how Site Editor > Pages works as well.

It looks like the only discrepancy is whether published post get confirmation or not.

**Comment by jameskoster** [comment]
Showing the confirmation is aligned with the [destructive actions guidance](https://wordpress.github.io/gutenberg/?path=/docs/design-system-patterns-destructive-actions--docs&globals=dsCornerRadius:subtle). The thinking behind that guidance was that users should either understand (and confirm) the impact of a destructive change up front, or be able to easily undo it quickly. The former was chosen because there's no de facto location for notices in the site editor, and adding one is a big project! 

Snackbar confirmations (with undo action) were another option, and would be a bit closer to the wp-admin experience, but we didn't want to more snackbars given their inherent a11y issues. I agree consistency is important, but it's a bit unclear how to get there in this instance given the constraints.

**Comment by afercia** [comment]
> Showing the confirmation is aligned with the [destructive actions guidance](https://wordpress.github.io/gutenberg/?path=/docs/design-system-patterns-destructive-actions--docs&globals=dsCornerRadius:subtle).

Thanks for your feedback. I have a few questions:

1
I see you linked to a documentation published on the Gutenberg storybook. Why that documentation isn't published on Make? 

2
Can you please point me to the relevant public discussion where the destructive actions guidance has been discussed? 

3
Are you aware there is a Trac ticket Core-49714 that was opened 6 years ago specifically about destructive actions and that still needs to be addressed? Why the process that led to elaborating the destructive actions guidance didn't happen on that Trac ticket? Any change that impacts Core should be discussed on Trac.

4
I'm not sure you replied to my concern about consistency. I'm not opposed to using a confirmation step when moving to Trash (though it doesn't make much sense to me). But if that stays in Gutenberg than it needs to be changed everywhere in WordPress. Including Core.

Users use WordPress and they don't care whether something comes from Core or Gutenberg.

They need a consistent user experience. I'm not sure showing confirmation in some parts of the WordPress UI and not showing them in other parts contributes to provide users with the best user experience. That really sounds a 'random' pattern. I'd like to stronlgy recommend again to take ownership and make sure the destructive actions guidance is applied consistently throughout the entire WordPress UI.

Cc @joedolson for the impact on Core-49714
